packages/xo-web/src/xo-app/vm/tab-general.js

```diff
@@ -48,13 +48,17 @@ export default connectStore(() => {
 
   return {
     lastShutdownTime: createGetVmLastShutdownTime(),
+    tasks: createGetObjectsOfType('task').filter((_, { vm }) => task =>
+      task.status === 'pending' && task.$ref in vm.current_operations
+    ),
```

Comment:
I believe it should be more performant to do a pick:

```js
createGetObjectsOfType('task')
  .pick(_, { vm } => Object.keys(vm.current_operations))
  .filter([ task => task.status === 'pending' ])
```

Reply:
`vm.current_operations`'s keys are refs though, not IDs.

Reply:
True, true.